runtime/queries/sql/highlights.scm

```diff
@@ -1,343 +1,391 @@
+(invocation
+  (object_reference
+    name: (identifier) @function.method))
 
 [
+  (keyword_gist)
   (keyword_btree)
   (keyword_hash)
-  (keyword_gist)
   (keyword_spgist)
   (keyword_gin)
   (keyword_brin)
-
-  (cast)
-  (group_concat)
-  (invocation)
+  (keyword_array)
 ] @function.builtin
-  
-(table_reference
-  name: (identifier) @namespace)
 
-(relation
-  table_alias: (identifier) @variable.parameter)
-  
-(field
+(object_reference
   name: (identifier) @variable.other.member)
-  
+
+(relation
+  alias: (identifier) @variable.parameter)
+
 (field
-  table_alias: (identifier) @variable.parameter
   name: (identifier) @variable.other.member)
 
-(comment) @comment
+(term
+  alias: (identifier) @variable.parameter)
 
-[
-  "("
-  ")"
-] @punctuation.bracket
+(term
+   value: (cast
+    name: (keyword_cast) @function.builtin
+    parameter: [(literal)]?))
 
-[
-  ";"
-  ","
-  "."
-] @punctuation.delimiter
+(literal) @string
+(comment) @comment.line
+(marginalia) @comment.block
 
-(binary_expression
-  operator: _ @operator)
+((literal) @constant.numeric.integer
+ (#lua-match? @constant.numeric.integer "^%d+$"))
```

Comment:
```suggestion
 (#match? @constant.numeric.integer "^\d+$"))
```

`#lua-match?` is nvim-specific. We use just the ones tree-sitter provides in the rust bindings https://tree-sitter.github.io/tree-sitter/using-parsers#predicates (though there are some there like `#any-of?` that we don't implement yet because we're waiting on a tree-sitter release that adds them)

Reply:
Also resolved. Thanks for your patience, I'm new to this tree-sitter world so don't know many of the keywords/possibilities yet, but it looks interesting